### PR TITLE
Fix races in NodeConfig disk test

### LINKED ISSUE #2093: NodeSetup disk test is racy and fails all future tests
There are several races in `Node Setup should make RAID0 array out of loop devices, format it to XFS, and mount at desired location out of one loop device [Serial]` which even prevent the test to be run on some setup entirely.

### Reusing devices from previous iterations
```
Node Setup should make RAID0 array out of loop devices, format it to XFS, and mount at desired location [It] out of one loop device [Serial]
/home/dev/dev/go/src/github.com/scylladb/scylla-operator/test/e2e/set/nodeconfig/nodeconfig_disksetup.go:270

  [FAILED] %!(EXTRA string=)
  Unexpected error:
      <exec.CodeExitError>: 
      command terminated with exit code 1
      {
          Err: <*errors.errorString | 0xc000c06810>{
              s: "command terminated with exit code 1",
          },
          Code: 1,
      }
  occurred
  In [It] at: /home/dev/dev/go/src/github.com/scylladb/scylla-operator/test/e2e/set/nodeconfig/nodeconfig_disksetup.go:201 @ 08/26/24 12:20:28.749
```

### Not retrying 409
```
Node Setup should make RAID0 array out of loop devices, format it to XFS, and mount at desired location [It] out of three loop devices [Serial]
/home/dev/dev/go/src/github.com/scylladb/scylla-operator/test/e2e/set/nodeconfig/nodeconfig_disksetup.go:271

  [FAILED] Unexpected error:
      <*errors.StatusError | 0xc000aae820>: 
      Operation cannot be fulfilled on nodeconfigs.scylla.scylladb.com "cluster": the object has been modified; please apply your changes to the latest version and try again
      {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "Operation cannot be fulfilled on nodeconfigs.scylla.scylladb.com \"cluster\": the object has been modified; please apply your changes to the latest version and try again",
              Reason: "Conflict",
              Details: {
                  Name: "cluster",
                  Group: "scylla.scylladb.com",
                  Kind: "nodeconfigs",
                  UID: "",
                  Causes: nil,
                  RetryAfterSeconds: 0,
              },
              Code: 409,
          },
      }
  occurred
  In [It] at: /home/dev/dev/go/src/github.com/scylladb/scylla-operator/test/e2e/set/nodeconfig/nodeconfig_disksetup.go:255 @ 08/26/24 12:21:18.539
```

### Cleanup not working (all the rest isn't cleaned up anyways)
```
Node Setup should make RAID0 array out of loop devices, format it to XFS, and mount at desired location [It] out of three loop devices [Serial]
/home/dev/dev/go/src/github.com/scylladb/scylla-operator/test/e2e/set/nodeconfig/nodeconfig_disksetup.go:282

  [FAILED] %!(EXTRA string=mdadm: Cannot get exclusive access to /host/dev/md108:Perhaps a running process, mounted filesystem or active volume group?
  )
  Unexpected error:
      <exec.CodeExitError>: 
      command terminated with exit code 1
      {
          Err: <*errors.errorString | 0xc000c54140>{
              s: "command terminated with exit code 1",
          },
          Code: 1,
      }
  occurred
  In [It] at: /home/dev/dev/go/src/github.com/scylladb/scylla-operator/test/e2e/set/nodeconfig/nodeconfig_disksetup.go:231 @ 08/26/24 12:32:37.414
```

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR fixes several races I have hit when running the NodeConfig disk setup test locally.

It also isolates the device names so a previously failed test doesn't bork all future tests.

**Which issue is resolved by this Pull Request:**
Resolves #2093

### CODE REVIEW

File: test/e2e/set/nodeconfig/nodeconfig_disksetup.go [hunk 1]
```diff
@@ -281,8 +282,18 @@ var _ = g.Describe("Node Setup", framework.Serial, func() {
...
+			var freshNC, updatedNC *scyllav1alpha1.NodeConfig
+			freshNC, err = f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs().Get(ctx, nc.Name, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			freshNC.Spec.LocalDiskSetup = nil
+			updatedNC, err = f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs().Update(ctx, freshNC, metav1.UpdateOptions{})
+			if err == nil {
+				nc = updatedNC
+			}
+			return err
```
Comment (rzetelskik) [inline test/e2e/set/nodeconfig/nodeconfig_disksetup.go:295]: nit: imo it's clearer to return err explicitly, but we don't have a coding convention yet so feel free to ignore

**Review by rzetelskik — APPROVED** [review]
one optional nit, lgtm
/assign zimnx

**Review by zimnx — APPROVED** [review]
/lgtm

/hold for Kacper's comment, feel free to remove it

File: test/e2e/set/nodeconfig/nodeconfig_disksetup.go (hunk 1 above)
Comment (tnozicka) [reply test/e2e/set/nodeconfig/nodeconfig_disksetup.go:295]: I think this is a bit special from the regular cases - you want to return the unmodified error whether it is nil an non-nil and no code can follow up after that.

Comment (rzetelskik) [reply test/e2e/set/nodeconfig/nodeconfig_disksetup.go:295]: > I am also not exactly sure what's proposed here? I assumed this was return nil

Yup, I meant a block for err != nil and return nil later.

> you want to return the unmodified error whether it is nil or non-nil and no code can follow up after that

fair enough, thanks

**Comment by tnozicka** [comment]
conversation on the nit got resolved in https://github.com/scylladb/scylla-operator/pull/2094#discussion_r1732891969
/hold cancel

**Pull request merged**